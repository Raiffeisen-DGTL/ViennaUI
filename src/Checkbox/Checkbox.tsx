import React, {
    useState,
    useCallback,
    Ref,
    forwardRef,
    FC,
    MouseEvent,
    ChangeEventHandler,
    FocusEventHandler,
} from 'react';
import { useControlState } from '@fcc/react-use';
import { Box, PropsBox, Content, Icon, IconBox, Input } from './Checkbox.styles';
import { ResponsiveProp, Breakpoints } from '../Utils/responsiveness';
import { BoxStyled } from '../Utils/styled';

export interface CheckboxProps<B = Breakpoints>
    extends Omit<BoxStyled<typeof Box, PropsBox>, 'onChange' | 'onFocus' | 'onBlur'> {
    size?: ResponsiveProp<'s' | 'm' | 'l', B>;
    indeterminate?: boolean;
    invalid?: boolean;
    active?: boolean;
    disabled?: boolean;
    checked?: boolean;
    defaultChecked?: boolean;
    name?: string;
    onChange?: ChangeEventHandler<HTMLInputElement>;
    onBlur?: FocusEventHandler<HTMLInputElement>;
    onFocus?: FocusEventHandler<HTMLInputElement>;
}

const fragment = {
    checked: 'M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z',
    indeterminate: 'M4 7h8v2H4V7z',
    empty: '',
};

const stopPropagation = (e: MouseEvent) => e.stopPropagation();

function CheckboxInternal<B = void>(
    {
        name,
        checked = false,
        invalid,
        indeterminate = false,
        size = 'm',
        disabled,
        tabIndex,
        defaultChecked,
        children,
        active: activeProps,
        onChange: onChangeProp,
        onBlur: onBlurProp,
        onFocus: onFocusProp,
        ...attrs
    }: CheckboxProps<B extends void ? Breakpoints : B>,
    ref: Ref<HTMLInputElement>
) {
    const [value, setValue] = useControlState({
        value: checked,
        defaultValue: defaultChecked,
        defaultStateValue: false,
    });

    const [active, setActive] = useState(false);

    const onBlur = useCallback<FocusEventHandler<HTMLInputElement>>(
        (e) => {
            if (!disabled) {
                setActive(false);
                onBlurProp?.(e);
            }
        },
        [onBlurProp, disabled]
    );

    const onFocus = useCallback<FocusEventHandler<HTMLInputElement>>(
        (e) => {
            if (!disabled) {
                setActive(true);
                onFocusProp?.(e);
            }
        },
        [onFocusProp, disabled]
    );

    const onChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
        (e) => {
            if (!disabled) {
                setValue(e.currentTarget.checked);
                onChangeProp?.(e);
            }
        },
        [setValue, onChangeProp, disabled]
    );

    return (
        <Box {...(attrs as {})} $disabled={disabled}>
            <IconBox>
                <Input
                    ref={ref}
                    type='checkbox'
                    checked={value || indeterminate}
                    name={name}
                    tabIndex={tabIndex}
                    aria-invalid={!!invalid}
                    disabled={disabled}
                    onChange={onChange}
                    onBlur={onBlur}
                    onFocus={onFocus}
                />
                <Icon
                    xmlns='http://www.w3.org/2000/svg'
                    viewBox='0 0 16 16'
                    $size={size}
                    $disabled={disabled}
                    $checked={value}
                    $invalid={invalid}
                    $indeterminate={indeterminate}
                    $active={activeProps ?? active}
                    focusable='false'
                    tabIndex={-1}
                    onClick={stopPropagation}>
                    <path
                        fill='currentColor'
                        d={fragment[indeterminate ? 'indeterminate' : value ? 'checked' : 'empty']}
                    />
                </Icon>
            </IconBox>
            {children && (
                <Content $size={size} onClick={stopPropagation}>
                    {children}
                </Content>
            )}
        </Box>
    );
}

export const Checkbox = forwardRef(CheckboxInternal) as <B = Breakpoints>(
    p: CheckboxProps<B> & { ref?: Ref<HTMLInputElement> }
) => JSX.Element;

(Checkbox as FC).displayName = 'Checkbox';
