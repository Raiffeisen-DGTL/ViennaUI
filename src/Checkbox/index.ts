export { Checkbox, type CheckboxProps } from './Checkbox';
