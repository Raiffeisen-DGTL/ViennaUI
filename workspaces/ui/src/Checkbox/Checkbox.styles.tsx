import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const checkbox = getPresets('checkbox', {
    base: null,
    size: 'size',
    hover: null,
    disabled: null,
    error: null,
    checked: null,
});

const label = getPresets('checkbox.label', {
    base: null,
    size: 'size',
    disabled: null,
});

export const IconBox = styled.span`
    position: relative;
    align-self: flex-start;
`;

export const Icon = styled.svg<{
    size: any;
    disabled: any;
    checked: any;
    indeterminate: any;
    invalid: any;
    active: any;
}>`
    display: block;
    position: relative;
    flex: 0 0 auto;
    box-sizing: border-box;
    color: transparent;
    transition: all .2s;

    ${checkbox.base}

    ${checkbox.size}

    ${({ checked, indeterminate }): any =>
        (checked || indeterminate) &&
        css`
            ${checkbox.checked}
        `}

    ${({ disabled }): any =>
        disabled &&
        css`
            ${checkbox.disabled}
        `}

    ${({ invalid }): any =>
        invalid &&
        css`
            ${checkbox.error}
        `}

    ${({ active }): any =>
        active &&
        css`
            ${checkbox.hover}
        `}
`;

export const Content = styled.span<{ size: any }>`
    ${label.base}
    ${label.size}
`;

export const Box = styled.label<{ disabled: any }>`
    padding: 2px 0;
    display: inline-flex;
    align-items: center;
    transition: all 0.2s;

    &:hover ${Icon} {
        ${checkbox.hover}
        ${({ disabled }): any =>
            disabled &&
            css`
                ${checkbox.disabled}
            `}
    }

    ${Content} {
        ${({ disabled }): any =>
            disabled &&
            css`
                ${label.disabled}
            `}
    }
`;

export const Input = styled.input`
    position: absolute;
    top: 0;
    left: 0;
    opacity: 0;
    width: 100%;
    height: 100%;
    margin: 0;
    z-index: -1;
`;
