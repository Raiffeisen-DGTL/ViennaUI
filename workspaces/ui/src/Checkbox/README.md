# Checkbox

Компонент для выбора нескольких значений из множества

## Импорт

```
import { Checkbox } from 'vienna-ui';
```

## Свойства / Props

Свойства наследуются от [HTMLAttributes<HTMLLabelElement>](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L1746)

| Prop | Type | Default | Description |
| --- | --- | --- | --- |
| children | ReactNode | false | Отображаемый лейбл |
| ref | ((instance: HTMLInputElement \| null) => void) \| RefObject \| null \| undefined | false | Сcылка на нативный элемент input, доступна после отрисовки |
| size | "s" \| "m" \| "l" \| undefined | "m" | Размеры |
| invalid | boolean \| undefined | false | Компонент отображается как ошибочный если true |
| indeterminate | boolean \| undefined | false | Частично выбранный чекбокс. Нельзя использовать совместно с checked |
| onBlur | CheckboxEvent<FocusEvent<HTMLInputElement>> \| undefined | false | Обработчик события при потере фокуса компонентом |
| onFocus | CheckboxEvent<FocusEvent<HTMLInputElement>> \| undefined | false | Обработчик события при получении фокуса компонентом |
| onChange | CheckboxEvent<FocusEvent<HTMLInputElement>> \| undefined | false | Обработчик события при клике на компонент |
| checked | boolean \| undefined | false | Статус выбранного значения |
| disabled | boolean \| undefined | false |
| name | string \| undefined | false |

## Использование

В качестве дочернего элемента передается label для checkbox.

> Компонент является контролируемым, то есть чтобы отобразить чтобы отобразить выбранные элементы, необходимо получить ее значение через обработчик `onChange` и прокинуть проверку в `checked`.

```jsx
{
    () => {
        const [state, setState] = React.useState({ checked: false, checked2: true, indeterminate: true });
        return (
            <Checkbox checked={state.checked} onChange={(e, { value }) => setState({ ...state, checked: value })}>
                Checkbox
            </Checkbox>
        );
    };
}
```

## Выбранный элемент

##### Свойство `checked`

```
<Checkbox checked>
    Content
</Checkbox>
```

## Размеры

##### Свойство `size`

Компонент имеет стандартные размеры `s`, `m` (по умолчанию) и `l`.

## Состояния

##### Свойство `disabled`, `invalid`

Компонент имеет состояние `disabled` и `invalid`

```jsx
<Checkbox disabled />
<Checkbox invalid />
```

## Частично выбранный чекбокс

##### Свойство `indeterminate`

Частично выбранный чекбокс. Применяется если в интерфейсе необходимо использование группы чекбоксов. Если часть из них выбрана, то применяется состояние частично выбранного чекбокса у родительского компонента.

```jsx
<Checkbox indeterminate>Label</Checkbox>
```
