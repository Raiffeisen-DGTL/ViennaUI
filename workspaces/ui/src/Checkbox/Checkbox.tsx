import React, { useState, useCallback, HTMLAttributes } from 'react';
import { Box, Content, Icon, IconBox, Input } from './Checkbox.styles';

export type CheckboxEvent<T> = (event: T, data?: { name?: string; value: boolean }) => void;
export interface CheckboxProps
    extends Omit<HTMLAttributes<HTMLLabelElement>, 'size' | 'onChange' | 'onBlur' | 'onFocus'> {
    /** Отображаемый лейбл */
    children?: React.ReactNode;
    /** Сcылка на нативный элемент input, доступна после отрисовки */
    ref?: React.Ref<HTMLInputElement>;
    /** Размеры */
    size?: 's' | 'm' | 'l';
    /** Имя компонента (важно для группировки) */
    name?: string;
    /** Состояние компонента */
    checked?: boolean;
    /** Компонент неактивен если true */
    disabled?: boolean;
    /** Компонент отображается как ошибочный если true */
    invalid?: boolean;
    /** Состояние компонента (тире если true) нельзя использовать совместно с checked */
    indeterminate?: boolean;
    /** Обработчик события при потере фокуса компонентом  */
    onBlur?: CheckboxEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при получении фокуса компонентом  */
    onFocus?: CheckboxEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при клике на компонент  */
    onChange?: CheckboxEvent<React.ChangeEvent<HTMLInputElement>>;
}

const fragment = {
    checked: 'M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z',
    indeterminate: 'M4 7h8v2H4V7z',
    empty: '',
};

export const Checkbox: React.FC<CheckboxProps> = React.forwardRef(
    (props: React.PropsWithChildren<Omit<CheckboxProps, 'ref'>>, ref: React.Ref<HTMLInputElement>): JSX.Element => {
        const {
            children,
            name,
            checked = false,
            invalid,
            indeterminate = false,
            size = 'm',
            disabled,
            tabIndex,
            onChange,
            onBlur,
            onFocus,
            ...attrs
        } = props;

        const [active, setActive] = useState(false);

        const handleEvent = useCallback(
            (callback: CheckboxEvent<any> | undefined) => {
                return (event: React.FormEvent<HTMLInputElement>) => {
                    if (event.type === 'blur') {
                        setActive(false);
                    }

                    if (!disabled) {
                        if (event.type === 'focus') {
                            setActive(true);
                        }
                        if (typeof callback === 'function') {
                            callback(event, { name, value: event.currentTarget.checked });
                        }
                    }
                };
            },
            [disabled, name]
        );

        return (
            <Box disabled={disabled} {...attrs}>
                <IconBox>
                    <Input
                        ref={ref}
                        type='checkbox'
                        checked={checked || indeterminate}
                        name={name}
                        tabIndex={tabIndex}
                        aria-invalid={!!invalid}
                        onChange={handleEvent(onChange)}
                        onBlur={handleEvent(onBlur)}
                        onFocus={handleEvent(onFocus)}
                    />
                    <Icon
                        xmlns='http://www.w3.org/2000/svg'
                        viewBox='0 0 16 16'
                        size={size}
                        focusable='false'
                        tabIndex={-1}
                        disabled={disabled}
                        checked={checked}
                        invalid={invalid}
                        indeterminate={indeterminate}
                        active={active}>
                        <path
                            fill='currentColor'
                            d={fragment[indeterminate ? 'indeterminate' : checked ? 'checked' : 'empty']}
                        />
                    </Icon>
                </IconBox>
                {children && <Content size={size}>{children}</Content>}
            </Box>
        );
    }
);

Checkbox.defaultProps = {
    size: 'm',
};

Checkbox.displayName = 'Checkbox';
export default Checkbox;
