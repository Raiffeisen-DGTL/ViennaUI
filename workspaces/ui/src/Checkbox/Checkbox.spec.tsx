import React from 'react';
import Checkbox, { CheckboxProps } from './Checkbox';

test('Checkbox', () => {
    const snap = snapshot.render([<Checkbox key='1' />, <Checkbox key='2' checked />]);
    expect(snap).toMatchSnapshot();
});

test('Checkbox w/ label', () => {
    const snap = snapshot.render([
        <Checkbox key='1'>Label</Checkbox>,
        <Checkbox key='2' checked>
            Label
        </Checkbox>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Checkbox w/ name', () => {
    const snap = snapshot.render([
        <Checkbox key='1' name='Checkbox1' />,
        <Checkbox key='2' name='Checkbox2' checked>
            Label
        </Checkbox>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Checkbox w/ disabled', () => {
    const snap = snapshot.render([
        <Checkbox key='1' name='Checkbox1' disabled />,
        <Checkbox key='2' name='Checkbox2' disabled>
            Label
        </Checkbox>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Checkbox w/ checked and disabled', () => {
    const snap = snapshot.render([
        <Checkbox key='1' name='Checkbox1' disabled checked />,
        <Checkbox key='2' name='Checkbox2' disabled checked>
            Label
        </Checkbox>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Checkbox w/ invalid', () => {
    const snap = snapshot.render([
        <Checkbox key='1' name='Checkbox1' invalid />,
        <Checkbox key='2' name='Checkbox2' invalid>
            Label
        </Checkbox>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Checkbox w/ size', () => {
    const sizes: CheckboxProps['size'][] = ['s', 'm', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(<Checkbox size={size}>Label</Checkbox>);
        expect(snap).toMatchSnapshot(size);
    });
});

test('Checkbox w/ any props', () => {
    const props = {
        className: 'className',
        id: 'id',
        title: 'title',
        onClick: () => 'Click',
    };

    const component = (
        <Checkbox {...props} name='name'>
            Label
        </Checkbox>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});

test('Checkbox w/ onBlur()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: false } };

    const snap = snapshot.shallow(
        <Checkbox name='checkbox' onBlur={fn}>
            Label
        </Checkbox>
    );

    snap.childAt(0).childAt(0).simulate('blur', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'checkbox', value: false });
});

test('Checkbox w/ onFocus()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: false } };

    const snap = snapshot.shallow(
        <Checkbox name='checkbox' onFocus={fn}>
            Label
        </Checkbox>
    );

    snap.childAt(0).childAt(0).simulate('focus', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'checkbox', value: false });
});

test('Checkbox w/ onChange()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Checkbox name='checkbox' onChange={fn}>
            Label
        </Checkbox>
    );

    snap.childAt(0).childAt(0).simulate('change', event);
    expect(snap).toMatchSnapshot();

    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'checkbox', value: true });
});

test('Checkbox w/ onChange() disabled', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Checkbox name='checkbox' disabled onChange={fn}>
            Label
        </Checkbox>
    );

    snap.childAt(0).childAt(0).simulate('change', event);
    expect(fn).not.toHaveBeenCalled();
    expect(fn).not.toHaveBeenCalledWith(event, { name: 'checkbox', value: true });
});
