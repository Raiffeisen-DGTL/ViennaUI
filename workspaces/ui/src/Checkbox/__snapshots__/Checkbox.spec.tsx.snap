// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 1`] = `
Array [
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 jkBeob"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
  </label>,
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        checked=""
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        type="checkbox"
      />
      <svg
        checked=""
        class="Checkboxstyles__Icon-sc-7b1pnz-1 fwBqJi"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z"
          fill="currentColor"
        />
      </svg>
    </span>
  </label>,
]
`;

exports[`Checkbox w/ any props 1`] = `
<label
  class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw className"
  id="id"
  title="title"
>
  <span
    class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
  >
    <input
      aria-invalid="false"
      class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
      name="name"
      type="checkbox"
    />
    <svg
      class="Checkboxstyles__Icon-sc-7b1pnz-1 jkBeob"
      focusable="false"
      tabindex="-1"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d=""
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
  >
    Label
  </span>
</label>
`;

exports[`Checkbox w/ checked and disabled 1`] = `
Array [
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 fMqNTb"
    disabled=""
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        checked=""
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox1"
        type="checkbox"
      />
      <svg
        checked=""
        class="Checkboxstyles__Icon-sc-7b1pnz-1 iWUwBx"
        disabled=""
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z"
          fill="currentColor"
        />
      </svg>
    </span>
  </label>,
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 fMqNTb"
    disabled=""
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        checked=""
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox2"
        type="checkbox"
      />
      <svg
        checked=""
        class="Checkboxstyles__Icon-sc-7b1pnz-1 iWUwBx"
        disabled=""
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Checkbox w/ disabled 1`] = `
Array [
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 fMqNTb"
    disabled=""
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox1"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 lmzwuE"
        disabled=""
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
  </label>,
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 fMqNTb"
    disabled=""
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox2"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 lmzwuE"
        disabled=""
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Checkbox w/ invalid 1`] = `
Array [
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="true"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox1"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 iCpjgk"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
  </label>,
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="true"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox2"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 iCpjgk"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Checkbox w/ label 1`] = `
Array [
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 jkBeob"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
    >
      Label
    </span>
  </label>,
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        checked=""
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        type="checkbox"
      />
      <svg
        checked=""
        class="Checkboxstyles__Icon-sc-7b1pnz-1 fwBqJi"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Checkbox w/ name 1`] = `
Array [
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox1"
        type="checkbox"
      />
      <svg
        class="Checkboxstyles__Icon-sc-7b1pnz-1 jkBeob"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d=""
          fill="currentColor"
        />
      </svg>
    </span>
  </label>,
  <label
    class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
  >
    <span
      class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
    >
      <input
        aria-invalid="false"
        checked=""
        class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
        name="Checkbox2"
        type="checkbox"
      />
      <svg
        checked=""
        class="Checkboxstyles__Icon-sc-7b1pnz-1 fwBqJi"
        focusable="false"
        tabindex="-1"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.207 5.207l-7 7a1 1 0 01-1.414 0l-4-4 1.414-1.414L6.5 10.086l6.293-6.293 1.414 1.414z"
          fill="currentColor"
        />
      </svg>
    </span>
    <span
      class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Checkbox w/ onChange() 1`] = `
<Checkboxstyles__Box>
  <Checkboxstyles__IconBox>
    <Checkboxstyles__Input
      aria-invalid={false}
      checked={false}
      name="checkbox"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <Checkboxstyles__Icon
      active={false}
      checked={false}
      focusable="false"
      indeterminate={false}
      size="m"
      tabIndex={-1}
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d=""
        fill="currentColor"
      />
    </Checkboxstyles__Icon>
  </Checkboxstyles__IconBox>
  <Checkboxstyles__Content
    size="m"
  >
    Label
  </Checkboxstyles__Content>
</Checkboxstyles__Box>
`;

exports[`Checkbox w/ size: l 1`] = `
<label
  class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
>
  <span
    class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
  >
    <input
      aria-invalid="false"
      class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
      type="checkbox"
    />
    <svg
      class="Checkboxstyles__Icon-sc-7b1pnz-1 fevQNn"
      focusable="false"
      tabindex="-1"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d=""
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    class="Checkboxstyles__Content-sc-7b1pnz-2 jEfea-D"
  >
    Label
  </span>
</label>
`;

exports[`Checkbox w/ size: m 1`] = `
<label
  class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
>
  <span
    class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
  >
    <input
      aria-invalid="false"
      class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
      type="checkbox"
    />
    <svg
      class="Checkboxstyles__Icon-sc-7b1pnz-1 jkBeob"
      focusable="false"
      tabindex="-1"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d=""
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    class="Checkboxstyles__Content-sc-7b1pnz-2 brKHIJ"
  >
    Label
  </span>
</label>
`;

exports[`Checkbox w/ size: s 1`] = `
<label
  class="Checkboxstyles__Box-sc-7b1pnz-3 rWrVw"
>
  <span
    class="Checkboxstyles__IconBox-sc-7b1pnz-0 iLaMGm"
  >
    <input
      aria-invalid="false"
      class="Checkboxstyles__Input-sc-7b1pnz-4 hCnUJI"
      type="checkbox"
    />
    <svg
      class="Checkboxstyles__Icon-sc-7b1pnz-1 ktXewH"
      focusable="false"
      tabindex="-1"
      viewBox="0 0 16 16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d=""
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    class="Checkboxstyles__Content-sc-7b1pnz-2 ciTVxy"
  >
    Label
  </span>
</label>
`;
